Support generation of Job/CronJob resources (#27425)
quarkusio/quarkus: docs/src/main/asciidoc/deploying-to-openshift.adoc

quarkus.openshift.env.fields.foo=metadata.name
----

==== Changing the generated deployment resource

> Beside generating a `DeploymentConfig` resource, you can also choose to get either a `Deployment`, `StatefulSet`, or a `Job`, or a `CronJob` resource instead via `application.properties`:

Reviewer: ```suggestion
Besides generating a `DeploymentConfig` resource, you can also choose to generate either a `Deployment`, `StatefulSet`, or a `Job`, or a `CronJob` resource instead via `application.properties`:
```

quarkus.container-image.group=<project/namespace name>
----

===== Generating Job resources

> If you want to generate a Job resource, you need to add the following property via the `application.properties`:

Reviewer: ```suggestion
If you want to generate a Job resource, you need to add the following property to the `application.properties`:
```

[source,properties]
----
quarkus.openshift.deployment-kind=Job
----

> IMPORTANT: If you are using the Picocli extension, by default the Job resource will be generated.

Reviewer: ```suggestion
IMPORTANT: If you are using the Picocli extension, by default a Job resource will be generated.
```

> You can provide the arguments that will be used by the Kubernetes Job via the property `quarkus.openshift.arguments`. For example, adding the property `quarkus.openshift.arguments=A,B`.

Reviewer: ```suggestion
You can provide the arguments that will be used by the Kubernetes Job via the property `quarkus.openshift.arguments`. For example, by adding the property `quarkus.openshift.arguments=A,B`.
```

> Finally, the Kubernetes job will be launched every time that is installed in OpenShift. You can know more about how to run Kubernetes jobs in this https://kubernetes.io/docs/concepts/workloads/controllers/job/#running-an-example-job[link].

Reviewer: ```suggestion
Finally, the Kubernetes job will be launched every time it is installed in OpenShift. You can know more about how to run Kubernetes jobs in this https://kubernetes.io/docs/concepts/workloads/controllers/job/#running-an-example-job[document].
```

> You can configure the rest of the Kubernetes Job configuration using the properties under `quarkus.openshift.job.xxx` (see https://quarkus.io/guides/deploying-to-openshift#quarkus-openshift-openshift-config_quarkus.openshift.job.parallelism[link]).

Reviewer: ```suggestion
You can configure the rest of the Kubernetes Job configuration using the properties under `quarkus.openshift.job.xxx` (see the https://quarkus.io/guides/deploying-to-openshift#quarkus-openshift-openshift-config_quarkus.openshift.job.parallelism[Deploying to OpenShift guide] for more information).
```

===== Generating CronJob resources

> If you want to generate a CronJob resource, you need to add the following property via the `application.properties`:

Reviewer: ```suggestion
If you want to generate a CronJob resource, you need to add the following properties to the `application.properties`:
```

[source,properties]
----
quarkus.openshift.deployment-kind=CronJob
# Cron expression to run the job every hour
quarkus.openshift.cron-job.schedule=0 * * * *
----

> IMPORTANT: CronJob resources require the https://en.wikipedia.org/wiki/Cron[Cron] expression to specify when to launch the job via the property `quarkus.openshift.cron-job.schedule`. If not provide, the build will fail.

Reviewer: ```suggestion
IMPORTANT: CronJob resources require the https://en.wikipedia.org/wiki/Cron[Cron] expression to specify when to launch the job via the property `quarkus.openshift.cron-job.schedule`. If not provided, the build will fail.
```

> You can configure the rest of the Kubernetes CronJob configuration using the properties under `quarkus.openshift.cron-job.xxx` (see https://quarkus.io/guides/deploying-to-openshift#quarkus-openshift-openshift-config_quarkus.openshift.cron-job.parallelism[link]).

Reviewer: ```suggestion
You can configure the rest of the Kubernetes CronJob configuration using the properties under `quarkus.openshift.cron-job.xxx` (see the https://quarkus.io/guides/deploying-to-openshift#quarkus-openshift-openshift-config_quarkus.openshift.cron-job.parallelism[Deploying to OpenShift guide] for more information).
```
